src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -130,6 +147,18 @@ public BoundAttributeDescriptor Build()
             return descriptor;
         }
 
+        public void Reset()
```

Comment:
Just to note: It's not a full reset because the `_containingTypeName` will be the same. Only way to get around that fact would be enable a re-initialization of that type.

Just commenting; I think this is fine as is.